### PR TITLE
PDP Containers

### PR DESCRIPTION
**Pull request by fnhipster**
- [x] @alandana update `package.json` with new @dropins/storefront-pdp release package – files are currently manually added hence why Cypress and Build are failing in the CI but working locally

Test URLs:
Before:
https://develop-aem-boilerplate-commerce--hlxsites.aem.live/products/hollister-backyard-sweatshirt/MH05
https://develop--aem-boilerplate-commerce--hlxsites.aem.page/products/sprite-yoga-companion-kit/24-WG080
After:
* https://pdp-containers--aem-boilerplate-commerce--hlxsites.aem.live
* https://pdp-containers--aem-boilerplate-commerce--hlxsites.aem.live/products/hollister-backyard-sweatshirt/MH05
* https://pdp-containers--aem-boilerplate-commerce--hlxsites.aem.live/products/sprite-yoga-companion-kit/24-WG080

### CODE REVIEW

File: blocks/product-details/product-details.js [hunk 1]
```diff
@@ -5,23 +5,303 @@ import { initializers } from '@dropins/tools/initializer.js';
...
+  // Initialize Dropins
+  initializers.register(PDP.initialize, {
+    sku: getSkuFromUrl(),
+    langDefinitions,
+    acdl: true,
+    persistURLParams: true,
+  });
+
+  // Get Initialized product data
+  const product = await PDP.getFetchedProductData();
```
Comment (sirugh) [inline blocks/product-details/product-details.js:100]: What's this? I see we are changing the prefetch logic in this PR from the scripts.js file.

Comment (fnhipster) [reply blocks/product-details/product-details.js:100]: The Product Data gets now fetched earlier: the initializer instead of the render. This function gets you that data at any point.

File: scripts/scripts.js [hunk 2]
```diff
@@ -229,14 +229,8 @@ async function loadEager(doc) {
   let pageType = 'CMS';
   if (document.body.querySelector('main .product-details')) {
     pageType = 'Product';
-    const sku = getSkuFromUrl();
-    window.getProductPromise = getProduct(sku);
```
Comment (sirugh) [inline scripts/scripts.js:233]: I believe `getProduct` is no longer used. This function should be removed.

File: blocks/product-details/product-details.js (hunk 1 above)
Comment (sirugh) [reply blocks/product-details/product-details.js:100]: What do you mean earlier? It was already prefetched in `scripts.js`, which happens before any of the pdp block is run.

Comment (fnhipster) [reply blocks/product-details/product-details.js:100]: You mean the custom graphql query than then is passed to the dropin that then skip it's own query. Does that sound right to you? This was patched this way to fetch data before the render which is not longer the case because the fetch happens on initialization. Want it earlier, initialization could be moved to script but I say let's test if that's even needed anymore.

File: scripts/scripts.js (hunk 2 above)
Comment (fnhipster) [reply scripts/scripts.js:233]: It should be removed. It's only used on "product-details-custom" which is a different block.

File: blocks/product-details/product-details.js [hunk 3]
```diff
@@ -1,26 +1,254 @@
...
+  const $description = fragment.querySelector('.product-details__description');
+  const $attributes = fragment.querySelector('.product-details__attributes');
+
+  block.appendChild(fragment);
+
+  // Alert
+  let inlineAlert = null;
+
+  // Gallery (Mobile)
+  await PDPProvider.render(ProductGallery, {
```
Comment (sirugh) [inline blocks/product-details/product-details.js:123]: As discussed on the call earlier, each of these `PDPProvider.render` lines is now sync and waiting for the prior one to complete. If there are no internal dependencies between the containers, we may as well wrap all in a resolving Promise like.

```
await Promise.all([
  await PDPProvider.render(...),
  await PDPProvider.render(...),
  ..., 
]);
```

Comment (fnhipster) [reply blocks/product-details/product-details.js]: Done

File: blocks/product-details/product-details.js [hunk 4]
```diff
@@ -1,26 +1,270 @@
...
+  const optionsUIDs = getOptionsUIDsFromUrl();
+
+  const [product, labels] = await Promise.all([
+    PDP.fetchPDPData(sku, { optionsUIDs }),
+    fetchPlaceholders(),
+  ]);
+
+  // Initialize
+  initializers.register(PDP.initialize, {
+    sku,
```
Comment (alandana) [inline blocks/product-details/product-details.js:62]: So is the SKU now mandatory in order to render the PDP container?

File: blocks/product-details/product-details.js [hunk 5]
```diff
@@ -1,26 +1,270 @@
...
+    optionsUIDs,
```
Comment (alandana) [inline blocks/product-details/product-details.js:63]: You can now pass the optionsUIDs here, but also through the models (as before). What's the difference? Isn't this a bit redundant or potentially cause confusion?

File: blocks/product-details/product-details.js (hunk 4 above)
Comment (fnhipster) [reply blocks/product-details/product-details.js:62]: If you are using the new containers, you must give an SKU to the initializer. However, this is optional if you use `ProductDetails` to make it backward compatible with integrations like Bulk. 

You can see examples of these two working on the PDP's sandbox:
- Legacy: https://git.corp.adobe.com/magento-datalake/storefront-pdp/blob/dca0cd5643693293951a6678851c5861c3432b46/examples/html-host/index.html#L224
- New: https://git.corp.adobe.com/magento-datalake/storefront-pdp/blob/dca0cd5643693293951a6678851c5861c3432b46/examples/html-host/index2.html#L203

File: blocks/product-details/product-details.js (hunk 5 above)
Comment (fnhipster) [reply blocks/product-details/product-details.js:63]: It's a way to provide `optionsUIDs` you may want to have pre-selected on the page load without having to pass the whole `initialData`.

**Review by alandana — APPROVED** [review]

**Comment by fnhipster** [comment]
it's happening

**Comment by fnhipster** [comment]
Note: PSI check it is also failing on `develop` branch. Might be an issue related to the data collector script.

File: blocks/product-details/product-details.js [hunk 6]
```diff
@@ -105,14 +342,11 @@ function setMetaTags(product) {
     return;
   }
 
-  const price = product.priceRange
-    ? product.priceRange.minimum.final.amount
-    : product.price.final.amount;
+  const price = product.prices.final.minimumAmount ?? product.prices.final.amount;
```
Comment (alandana) [inline blocks/product-details/product-details.js:345]: @fnhipster why the change here?

**Review by alandana — APPROVED** [review]
